lib/pbench/agent/results.py

```diff
@@ -102,100 +122,78 @@ def make_result_tb(self):
-        tarball_md5 = Path(f"{tarball}.md5")
-        try:
-            hash_md5 = md5sum(tarball)
-            tarball_md5.write_text(f"{hash_md5} {tarball.name}\n")
-        except Exception:
-            self.logger.error("md5sum failed for {}, skipping", tarball)
-            try:
-                tarball.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error("error removing failed tar ball, {}", tarball)
-            try:
-                tarball_md5.unlink()
-            except OSError as exc:
-                if exc.errno != errno.ENOENT:
-                    self.logger.error(
-                        "error removing failed tar ball MD5, {}", tarball_md5
-                    )
-            sys.exit(1)
-
 
 class CopyResultTb:
+    """CopyResultTb - Copies tarball from agent to server with the help of
+            RestAPI PUT request
+    """
+
     chunk_size = 4096
 
-    def __init__(self, controller, tarball, config, logger):
+    def __init__(
+        self, controller: str, tarball: str, config: PbenchAgentConfig, logger: Logger
+    ):
         self.tarball = Path(tarball)
         if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
-        self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
+            raise FileNotFoundError(f"Tarball does not exist, '{self.tarball}'")
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
+        server_rest_url = config.get("results", "server_rest_url")
         self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
 
-    def read_in_chunks(self, file_object):
+    def read_in_chunks(self, file_object: IO) -> Iterator[AnyStr]:
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
-        files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
-        file_count = len(files)
-        if file_count != 2:
-            self.logger.error(
-                "(internal): unexpected file count, {}, associated with tarball, '{}'",
-                file_count,
-                self.tarball,
-            )
-            sys.exit(1)
-
-        with open(self.tarball_md5, "r") as md5fp:
-            md5sum = md5fp.read().split()[0]
-        filename = secure_filename(str(self.tarball))
-        headers = {"filename": filename, "Content-MD5": md5sum}
+    def copy_result_tb(self, token: str) -> None:
+        """copy_result_tb - copies tb from agent to configured server upload URL
+
+            Args
+                token -- an Authorization key which will allow the user to be
+                    authorized by the server.
```

Comment:
"an authorization key [...] to be authorized" is weak. Authorization token to identify the user to the server? Hmm...

Reply:
Perhaps, `token -- a previously generated token authorizing the use of the PUT API.`

Reply:
I changed it to "a key which will allow the user to be authenticated by the server."  Feel free to offer further suggestions.

Reply:
Isn't the token an authorization mechanism? I have a (valid) token, so that authorizes me to PUT.  I was allowed to get a token in the first place because I logged in and the server authenticated me.

Reply:
> Isn't the token an authorization mechanism? I have a (valid) token, so that authorizes me to PUT. I was allowed to get a token in the first place because I logged in and the server authenticated me.

Yes, but the token is generated by a login handshake and then presented to a completely separate interface (PUT) so that PUT can validate the user's authorization to complete that operation.

I still think the wording is a bit awkward; maybe something like "a token to establish that the caller is authorized to use the PUT method on behalf of an authenticated user" ... although that's awkward, too.

Reply:
You present a token with your PUT request; the token allows your identity to be authenticated, and it allows your authorization to be confirmed -- both of which are required for the request to proceed.

If the token is valid, that means that the request is coming from a specific, authentic user.  The token also either contains authorization information or it functions as a key to a database of authorizations -- either way, it allows the server to determine whether the request is authorized to proceed.

So, yes, the token is a means of demonstrating authorization, but it also provides authentication.

While authentication does not imply authorization, authorization rarely exists independent of authentication.

I can change it to something like, "a key which allows the server to authenticate the user and confirm whether the request is authorized".  Is that better?

Reply:
>  "a token to establish that the caller is authorized to use the PUT method on behalf of an authenticated user"

I like that.  I would wordsmith it as follows:  "a token which establishes that the caller is authorized to make the PUT request on behalf of a specific user".  (I would suggest that authorization implies authentication, so we don't need to mention that, here.)

Reply:
Changed.